hardware/simulation/icarus/Makefile

```diff
@@ -26,4 +26,4 @@ a.out: $(VSRC) $(VHDR)
 clean:
 	@rm -f *# *~ *.vcd a.out *_tb
 
-.PHONY: run clean
+.PHONY: a.out run clean
```

Comment:
.PHONY means the targets are always executed

why do you want to rebuild the simulation script a.out if it is already built?

It is built only if the prerequisites $(VSRC) $(VHDR) are newer (make checks the file dates). Otherwise, it doesn't need to build, only to run.

What would be useful is to move the .PHONY declaration to before the phony targets.

Reply:
This position of .PHONY is our new convention because I read somewhere it is important to do that.

Reply:
I don't see a problem to do make clean, that's what it exists 

PHONY targets may match existing files : you are telling the system that you want it rebuilt whether it exists or not

Reply:
For make sim VCD=1, the solution suggested allows a user to run only one target to clean and rebuild automatically and simulate.